### PR TITLE
662. Maximum Width of Binary Tree

### PR DESCRIPTION
**Pull request by kwang18**

### CODE REVIEW

**Comment by kwang18** [comment]
@tianheng

File: 662. Maximum Width of Binary Tree [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
+ * };
+ */
+//Time : O(n) Space : O(h), where n is number of nodes in the tree and h is height of tree
```
Comment (tianheng) [inline 662. Maximum Width of Binary Tree:10]: Give me a reason why you need h here

Comment (kwang18) [reply 662. Maximum Width of Binary Tree:10]: The height of the tree can vary from log n in its best case to n in its worst case. I understand we are talking about the worst case and it should be O(n).
I saw some people describe their algorithm complexity in this way before. I will no longer use this in future.

Comment (tianheng) [reply 662. Maximum Width of Binary Tree:10]: well, that depends, for example, people usually use average performance to judge quick sorting, not worst. It is ok you use h here, but I just want to know your reason.

**Pull request merged**